### PR TITLE
Feature/itsasu 682 overseas property business commencement

### PR DESCRIPTION
**Pull request by aug24**

### CODE REVIEW

File: app/views/agent/business/OverseasPropertyStartDate.scala.html [hunk 1]
```diff
@@ -16,36 +16,34 @@
...
-            label = Some(messages("agent.overseas.property.name.heading")),
-            content = Some(inputContent),
+            legendContent = messages("agent.overseas.property.line_1"),
+            hintText = Some(messages("agent.overseas.property.start_date.exampleStartDate")),
             parentForm = overseasPropertyStartDateForm,
-            hintText = Some(messages("agent.overseas.property.start_date.exampleStartDate"))
+            isPageHeading = false
         )
 
         @if(isEditMode) {
```
Comment (sb-dev) [inline app/views/agent/business/OverseasPropertyStartDate.scala.html:45]: The continue button helper can receive the `isEditMode` flag:
```
@continueButton(isEditMode = isEditMode)
```

File: app/views/agent/business/OverseasPropertyStartDate.scala.html [hunk 2]
```diff
@@ -16,36 +16,34 @@
...
-    @summaryErrorHelper(overseasPropertyStartDateForm)
+    <h1 class="govuk-heading-l">@messages("agent.overseas.property.name.heading")</h1>
 
     @form(action = postAction) {
 
         @dateHelper(
             field = overseasPropertyStartDateForm(startDate),
-            label = Some(messages("agent.overseas.property.name.heading")),
-            content = Some(inputContent),
+            legendContent = messages("agent.overseas.property.line_1"),
```
Comment (sb-dev) [inline app/views/agent/business/OverseasPropertyStartDate.scala.html:39]: It looks like the `legendContent` displays the text as bold unlike the original page.

File: app/views/agent/business/OverseasPropertyStartDate.scala.html [hunk 3]
```diff
@@ -16,36 +16,34 @@
 
 @import uk.gov.hmrc.play.views.html.helpers.form
```
Comment (sb-dev) [inline app/views/agent/business/OverseasPropertyStartDate.scala.html:17]: `@import uk.gov.hmrc.play.views.html.helpers.form` is not used anymore.

File: app/controllers/agent/business/OverseasPropertyStartDateController.scala [hunk 4]
```diff
@@ -45,25 +47,22 @@ class OverseasPropertyStartDateController @Inject()(val auditingService: Auditin
...
           (implicit request: Request[_]): Html = {
-    views.html.agent.business.overseas_property_start_date(
+    overseasPropertyStartDate(
       overseasPropertyStartDateForm = overseasPropertyStartDateForm,
-      postAction = controllers.agent.business.routes.OverseasPropertyStartDateController.submit(editMode = isEditMode),
-      isEditMode = isEditMode,
-      backUrl = backUrl(isEditMode, incomeSourceModel)
-    )
+      routes.OverseasPropertyStartDateController.submit(),
+      backLink,
```
Comment (sb-dev) [inline app/controllers/agent/business/OverseasPropertyStartDateController.scala:55]: It looks like you stop using the `backUrl` function?
```
  def backUrl(isEditMode: Boolean, incomeSourceModel: IncomeSourceModel): String = {
    if (isEditMode) {
      controllers.agent.routes.CheckYourAnswersController.show().url
    } else {
      (incomeSourceModel.selfEmployment, incomeSourceModel.ukProperty) match {
        case (_, true) => controllers.agent.business.routes.PropertyAccountingMethodController.show().url
        case (true, _) => appConfig.incomeTaxSelfEmploymentsFrontendUrl + "/client/details/business-accounting-method"
        case _ => controllers.agent.routes.IncomeSourceController.show().url
      }
    }
  }
```

File: app/views/agent/business/OverseasPropertyStartDate.scala.html (hunk 1 above)
Comment (aug24) [reply app/views/agent/business/OverseasPropertyStartDate.scala.html:45]: There's also a different param in the alternateText?  I don't think it adds anything.

**Review by sb-dev — APPROVED** [review]

**Pull request merged**